python/pyspark/pandas/accessors.py

```diff
@@ -335,14 +335,16 @@ def apply_batch(
         if not isinstance(func, FunctionType):
             assert callable(func), "the first argument should be a callable function."
             f = func
-            func = lambda *args, **kwargs: f(*args, **kwargs)
+            func = lambda *args, **kwargs: f(*args, **kwargs)  # noqa: E731
```

Comment:
Why is this still using lambda?

Reply:
Ah, otherwise, it complains that `func` cannot be redefined in mypy later.

Reply:
and it shouldn't have return type because it fails to properly infer the schema later with `infer_return_type`.

Reply:
let me add some comments.